carma_nav2_port_drayage_demo/include/carma_nav2_port_drayage_demo/port_drayage_demo.hpp

```diff
@@ -16,101 +16,104 @@
+    ENTER_PORT,
+    EXIT_PORT,
+    PORT_CHECKPOINT,
+    HOLDING_AREA,
+    DEFAULT_OPERATION,
+  };
+
+  /**
+    * \brief Standard constructor for OperationID
+    * \param op Operation enum associated with this object.
+    */
+  explicit OperationID(enum Operation op) : operation_enum_(op) {}
+
+  /**
+    * \brief Create OperationID using a string
+    * \param op_str String to used to create the Operation
+    */
+  explicit OperationID(std::string op_str);
+
+  /**
+    * \brief Getter function to obtain the Operation enum associated with this object.
+    * \return Operation enum associated with this object.
+    */
+  OperationID::Operation getOperationID() const;
+
+  /**
+    * \brief Function to convert a string to Operation
+    * \param op_str String to convert to an Operation
+    */
+  OperationID::Operation stringToOperation(std::string op_str) const;
+
+  /**
+    * \brief Function to convert this object's 'operation_enum_' to a human-readable string.
+    * \return A human-readable string representing this object's 'operation_enum_'.
+    */
+  std::string operationToString() const;
+
+  /**
+    * \brief Stream operator for this object.
+    */
+  friend std::ostream & operator<<(std::ostream & output, const OperationID & oid)
+  {
+    return output << oid.operationToString();
+  }
+
+  /**
+    * \brief Overloaded == operator for comparision with String objects.
+    */
+  friend bool operator==(const std::string & lhs, const OperationID & rhs)
+  {
+    return lhs == rhs.operationToString();
+  }
 
+private:
+  // Data member containing this object's Operation enum value
+  const Operation operation_enum_ = Operation::DEFAULT_OPERATION;
+};
 
 struct PortDrayageMobilityOperationMsg
 {
-    std::string cargo_id;
-    std::shared_ptr<OperationID> operation;
-    std::string current_action_id; // Identifier for the action this message is related to
-    double dest_longitude;  // Destination longitude for the carma vehicle
-    double dest_latitude;   // Destination latitude for the carma vehicle
-    double start_longitude; // Starting longitude of the carma vehicle
-    double start_latitude;  // Starting latitude of the carma vehicle
-    PortDrayageMobilityOperationMsg()
-    {
-      operation = std::shared_ptr<OperationID>(new OperationID{OperationID::Operation::ENTER_STAGING_AREA});
-    }
+  std::string cargo_id;
+  std::shared_ptr<OperationID> operation;
+  std::string current_action_id;  // Identifier for the action this message is related to
+  double dest_longitude;          // Destination longitude for the carma vehicle
+  double dest_latitude;           // Destination latitude for the carma vehicle
+  double start_longitude;         // Starting longitude of the carma vehicle
+  double start_latitude;          // Starting latitude of the carma vehicle
+  PortDrayageMobilityOperationMsg()
+  {
+    operation =
+      std::shared_ptr<OperationID>(new OperationID{OperationID::Operation::ENTER_STAGING_AREA});
```

Comment:
My initial though was to initialize this to enter staging area because that is the first operation we will receive in the final demo, but yeah I think it makes more sense to initialize this to default operation. Just made that change!